sktime/forecasting/adapters/_autots.py

```diff
@@ -0,0 +1,449 @@
+            )
+
+        self.forecaster_ = self._ModelClass(
+            model_list=model_list,
+            forecast_length=self._get_forecast_length(),
+            frequency=self.frequency,
+            prediction_interval=self.prediction_interval,
+            max_generations=self.max_generations,
+            no_negatives=self.no_negatives,
+            constraint=self.constraint,
+            ensemble=ensemble,
+            initial_template=self.initial_template,
+            random_seed=self.random_seed,
+            holiday_country=self.holiday_country,
+            subset=self.subset,
+            aggfunc=self.aggfunc,
+            na_tolerance=self.na_tolerance,
+            metric_weighting=metric_weighting,
+            drop_most_recent=self.drop_most_recent,
+            drop_data_older_than_periods=self.drop_data_older_than_periods,
+            num_validations=self.num_validations,
+            models_to_validate=self.models_to_validate,
+            validation_method=self.validation_method,
+            min_allowed_train_percent=self.min_allowed_train_percent,
+            remove_leading_zeroes=self.remove_leading_zeroes,
+            transformer_list=self.transformer_list,
+            transformer_max_depth=self.transformer_max_depth,
+            models_mode=self.models_mode,
+            max_per_model_class=self.max_per_model_class,
+            prefill_na=self.prefill_na,
+            introduce_na=self.introduce_na,
+            preclean=self.preclean,
+            model_interrupt=model_interrupt,
+            generation_timeout=self.generation_timeout,
+            current_model_file=self.current_model_file,
+            verbose=self.verbose,
+            n_jobs=self.n_jobs,
+        )
+        return self
+
+    def _convert_input_to_date(self, y):
+        """
+        Coerce y.index to pd.DatetimeIndex, for use by AutoAF.
+
+        Parameters
+        ----------
+            y (pd.DataFrame or None): The input data.
+
+        Returns
+        -------
+            pd.DataFrame or None: The input data with y.index coerced to
+                pd.DatetimeIndex, or None if y is None.
+        """
+        if y is None:
+            return None
+        elif type(y.index) is pd.PeriodIndex:
+            y = y.copy()
+            y.index = y.index.to_timestamp()
+        elif y.index.is_integer():
+            y = self._convert_int_to_date(y)
+        # else y is pd.DatetimeIndex as AutoAF expects, and needs no conversion
+        return y
+
+    def _convert_int_to_date(self, y):
+        """
+        Convert int to date, for use by AutoAF.
+
+        Parameters
+        ----------
+            y (pandas.Series): The input series containing integer values.
+
+        Returns
+        -------
+            pandas.Series: The input series with the index converted to dates.
+        """
+        y = y.copy()
+        idx_max = y.index[-1] + 1
+        int_idx = pd.date_range(start="2000-01-01", periods=idx_max, freq="D")
```

Comment:
I don't have a suggestion, but I don't think we should do this.

Reply:
well, we did almost the same in `Prophet`, if I remember correctly...